### DISCUSSION TITLE
Update on gene expression databases including pre-clinical and animal species

### DISCUSSION THREAD

**Discussion by HenrikCordes** [Open-Systems-Pharmacology/Forum#1096] [Announcement]
Dear open-systems-pharmacology community,

we are currently working on an update of the gene expression databases of PK-Sim.
To this end we're preparing a publication that will outline the database generation, the R-Code that creates them and the first release the of the databases. Ideally we could include them in one of the next releases and make the scripts part of the git repository, let's see.

It will still take a while, but nevertheless we wanted to keep you updated and a first glimpse of what you can expect within 2022.

As requested in previous posts #1125 #1069 #882 #864, the next line of gene expression databases will cover various animal species relevant in pre-clincial research, animal healthy and toxicology, as well as an update on the human data.
Currently we cover the following species:
**Human, Monkey** (Macaca mulatta & Macaca fascicularis), **Minipig, Dog, Rat, Mouse, Rabbit, Zebrafish, Cattle, Horse, Cat, Guinea Pig, Chicken, Turkey, Goat and Sheep**

All gene annotations are made with the biomaRt - Bioconductor package and are based on ensemble and NCBI gene identifiers.
For animal species, the databases include a link to the human homolog, such that you can query for example for a human CYP3A4 and the database suggests the animal homologs.

The source data is based generated from **RNAseq experiments** derived from **healthy primary tissue samples** gathered from public sources. No cell line or disease data is included. Note that the databases also do not include protein abundance information.

**I'm happy to present you the **beta versions of Rat and Rabbit. Feel free to check them out!**
Let me know if you find bugs or have any suggestions, I'm happy to include them in the next release**.

[GENEDB_rat_TPM_ONLY_BgeeRelease_14_0_2022-01-26.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/7945776/GENEDB_rat_TPM_ONLY_BgeeRelease_14_0_2022-01-26.zip)
[GENEDB_rabbit_TPM_ONLY_BgeeRelease_14_1_2022-01-26.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/7945778/GENEDB_rabbit_TPM_ONLY_BgeeRelease_14_1_2022-01-26.zip)

Kind regards,
Henrik

@sfrechen PS: Based on your request and suggestion that this will be most interesting for others,  I put together the human database.
So here it is, fresh out of the press: 
[GENEDB_human_ADME_ONLY_TPM_ONLY_BgeeRelease_14_0_2022-02-15.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8075131/GENEDB_human_ADME_ONLY_TPM_ONLY_BgeeRelease_14_0_2022-02-15.zip)

PSPS: Please note, this beta-version for human data contains **ADME genes only**  and is **based only on a fraction of the available data** for example not including data form the "Genotype-Tissue Expression (GTEx) Common Fund Project" and other viable source that might influence the overall expression pattern.

**Comment by msevestre** [comment]
This is fantastic! What an amazing contribution. Thank you.

**Comment by teutonicod** [comment]
Great contribution @HenrikCordes, thanks a lot!

**Comment by LarsKuepfer** [comment]
Excellent! Amazing work!!!

**Comment by tobiasK2001** [comment]
Yes, a great piece of work! Thanks Henrik

**Comment by CristianVU** [comment]
Thanks a lot! I will be looking forward to the dog database.

**Reply by HenrikCordes** [reply]
Hi @CristianVU ,

thank you for posting your request #1108  and your interest in the Dog gene expression DB.
As mentioned we are preparing the manuscript that will officially release the DBs.

In the mean time feel free to check out the beta version for the Dog:

[GENEDB_dog_ADME_ONLY_TPM_ONLY_BgeeRelease_14_1_2022-02-05.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/8008180/GENEDB_dog_ADME_ONLY_TPM_ONLY_BgeeRelease_14_1_2022-02-05.zip)

As mentioned above the DB is not reviewed yet and you should be careful when developing hypothesis based on this information.
If you finde something odd or have any suggestions to improve the DB I'm happy to take your comments and suggestions.

Best
Henrik

**Reply by CristianVU** [reply]
Thank you so much. Was there an update to the dog gene pool? Was there an official version released?
At the moment I am using the beta version posted on the forum. It's great !

**Comment by msevestre** [comment]
@HenrikCordes Yes. Can you create an issue in PKSim repositories for this specific request? I think we need to track this separately

**Reply by CristianVU** [reply]
Yes!! And follow new release about dog database !

**Comment by HenrikCordes** [comment]
Hi Community,

as this was requested during the PAGE, please find attatched the Sheep ADME gene expression DB:
[GENEDB_sheep_ADME_ONLY_TPM_ONLY_BgeeRelease_15_0_2022-07-23.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/9207135/GENEDB_sheep_ADME_ONLY_TPM_ONLY_BgeeRelease_15_0_2022-07-23.zip)

Kind regards,
Henrik

**Comment by RobinM92** [comment]
Hi @HenrikCordes ,

Not sure if this is the right place to ask, but is it possible that the expression for the rat's homologue of BCRP is missing? Should be gene Abcg3, and it is present in the Bgee database (https://www.bgee.org/gene/ENSRNOG00000030216/), but not in the GeneDb created from it (or I am missing it completely.)

Thanks again for this tremendous work!

**Reply by RobinM92** [reply]
It is present in the db as ENSRNOG00000030216 but no aliases such as BCRP, Abcg3,.. are present

**Reply by HenrikCordes** [reply]
Hi @RobinM92,

thank you for your patients and for using the pre-clinical species database and post the issue you came across! 

If I understood you correctly, you want to use the rat ortholog of the human BCRP gene in your model? 
Here is how I would go about it: 
1. Link the Rat DB to PK sim (in the utility tab)
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/c8bbac1c-80c1-4a26-a256-a023d93ebf5d)

2. Add an expression profiles for BCRP
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/69884840-5818-404b-9195-e78c3e860d8b)

3. Search the db for "BCRP" 
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/12f3be7b-89a0-4697-950b-a364fc1654d6)

4. Select & validate that you have the appropriate gene result.
Follow the link of the rat gene to NCBI
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/a80112a6-a50f-4f19-b71a-09497140668f)
Search for the human gene you want to have the ortholog 
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/3b897d42-0961-489e-9e69-95e75a6200e3)

5. Add the expression profile
![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/4817133/327908e4-69a5-45ed-a9dc-8e61564f204c)


From that exercise you can see that the rat version of the human BCRP (ABCG2) gene is the rat Abcg2. 
The orthologs you mention ENSRNOG00000030216 or ENSRNOG00000070774 share ~ 50 % sequence identity with humans. 
It is up to you to decide if the rat BRCP version ENSRNOG00000007041 or one of the other ENSRNOG00000030216 or ENSRNOG00000070774 genes are more representative / adequate for your modeling setting. Notably ENSRNOG00000070774  is not part of the bgee data, so you would have to choose from the other two. 

Let me know if this helps! 

Best
Henrik

**Reply by RobinM92** [reply]
Hi @HenrikCordes,

Thanks for your detailed response! This is indeed the workflow that I wanted to follow, however, looking for BCRP in the rat database points towards an unrelated enzyme (dcaf5):

![image](https://github.com/Open-Systems-Pharmacology/Forum/assets/30694339/098509f5-3442-498c-835f-77b1e2b15449)

Should I try downloading the rat database again or do you have another idea why this could occur?

All the best,
Robin

**Reply by HenrikCordes** [reply]
Hi @RobinM92,

I tried re-downloading the DB and, as you mentioned, that version does not contain the BCRP gene. 
Also, the latest packaging of the Rat ADME only database does not contain the BCRP gene.
Can be found here: https://github.com/Open-Systems-Pharmacology/Gene-Expression-Databases/releases/download/v3.0.0/GENEDB_rat_ADME_ONLY_TPM_ONLY_BgeeRelease_15_0_2023-08-03.zip

On my end I use the "full" database not only containing ADME genes: 
https://github.com/Open-Systems-Pharmacology/Gene-Expression-Databases/releases/download/v3.0.0/GENEDB_rabbit_ADME_ONLY_TPM_ONLY_BgeeRelease_15_0_2022-12-04.zip 

In the full DB you find the genes with the steps I outlined above. 

You can also use the published R code to fetch the latest data from Bgee and Annotation DB to set up the DB locally on your PC.
Let me know if this helps.

Best
Henrik

**Reply by RobinM92** [reply]
Hi @HenrikCordes,

Amazing, thanks so much for updating. Should I run the published R-code for the other preclinical species (mice, beagle, NHP) I'm looking at to be sure?

All the best,
Robin

**Reply by HenrikCordes** [reply]
Hi @RobinM92,

I might find time next week to update the other DBs on the Git.
In the mean time you can try to run the R-Code to set up the DBs locally. 
Would be a good test for my code if you try and we can make it work :-D 

If you face issues let me know. 

Best
Henrik

**Reply by RobinM92** [reply]
Hi @HenrikCordes,

It seems quite some packages used in that code are "not available for my version of R. (4.0.5), have you seen this before?

Kind regards,
Robin

**Reply by HenrikCordes** [reply]
Hi @RobinM92,

I used a different Version of R and probably also other package versions. 

For you convenience, I share with you the renv snapshot of my machine setting along with the code to set up the "zoo" :-D 
You can use the renv file to set up your environment with the same R and package versions. 

Let me know if this helps. 

Best
Henrik 

[Code.zip](https://github.com/Open-Systems-Pharmacology/Forum/files/12258004/Code.zip)

**Reply by RobinM92** [reply]
Hi @HenrikCordes ,

Thanks a lot! However, even when using the snapshot there are issues with the installation of some packages (e.g. fansi). I'll keep going at it, but if you would have the time to update the repository for mouse, dog and NHP in the meantime that would be incredible.

Kind regards,
Robin

**Reply by RobinM92** [reply]
@HenrikCordes, apologies to ping you again, but would it be possible to update the repository for these preclinical species? Thank you in advance!

**Reply by HenrikCordes** [reply]
Hi @RobinM92,
Could you progress in the mean time? 
Did you manage to install the needed R version and the R packages from the lock file?
In principal you should only have to run renv::restore() and the environment is set up.

Best
Henrik
